packages/docs/developer-resources/walkthroughs/web-dapp.md

```diff
@@ -0,0 +1,299 @@
+      {proposal.vote ? (
+        <span>{proposal.vote}</span>
+      ) : (
+        <div>
+          <button onClick={() => vote(proposal.id, VoteValue.Yes)}>Yes</button>
+          <button onClick={() => vote(proposal.id, VoteValue.No)}>No</button>
+        </div>
+      )}
+    </td>
+  </tr>
+)
+```
+
+## Best practices
+
+We've compiled a short list on best practices to follow when developing DApps. Following these will improve the end user experience and keep them more engaged with the Celo ecosystem. If you have any questions around these, feel free to [reach out on Discord](https://discord.gg/745Qntv), we're always there and happy to chat.
+
+### Last used address
+
+[@celo-tools/use-contractkit](https://github.com/celo-tools/use-contractkit) will remember the address a user last logged in with (via [browser LocalStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)). Use this to your advantage and allow your DApp to display the same data whether or not the user has connected their wallet. A good test is to refresh your DApp after connecting and see if anything changes. At the very most, buttons for interaction could be disabled, however it's preferable to prompt to connect the wallet on button click.
+
+Keeping the UI consistent by using the last connected address is a quick win we can have with DApps that make the experience using them closer to Web2, an experience more users will be familiar with.
+
+### Loading states
+
+Loading times are often the give away that an application is a Web3 DApp. Be liberal with loading screens and prioritise making animations smooth.
+
+Nothing is worse than a perpetually hanging screen that takes multiple seconds to become interactive. By showing a spinner it communicates to the user that things are happening, however slow they may be.
+
+This is often offset by the ability to index a blockchain and provide the data in a more accessible format (maybe a SQL database or behind a GraphQL API). As mentioned earlier we haven't covered that in this tutorial, however there's a lot of content on the web around DApp optimisation through prior state indexing.
+
+### Prerender what you can
+
+With modern static site generators we have amazing leverage over what gets computed server side and what the browser has to request and compute before rendering. If you're unable to index the blockchain before a client requests access to a page, consider loading the relevant data server side with a cache invalidated every hour or so.
+
+Next.js [getStaticProps](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation) comes to mind here as a great way to offload heavy computation to the server.
+
+### Showing numbers in wei vs. Celo
+
+Take this advice with a grain of salt as it really depends on how proficient your users are. At some point in most DApp users are going to need to deal with large numbers. It's up to you whether you display these in wei (1e18) or not.
```

Comment:
maybe instead of "proficient" something like "familiar with [cryptocurrencies/blockchain] your users are"?